### PR TITLE
Feat(Builder): Discord bots blocks

### PR DESCRIPTION
**Pull request by Bentlybro**
### Background

This is a super early version of discord bot blocks

### Changes 🏗️

Added a file ``discordblocks.py`` which contains the code, updated ``settings.py`` to add 
``discord_bot_token: str = Field(default="", description="Discord bot token")``

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Comment by Bentlybro** [comment]
test

File: rnd/autogpt_server/autogpt_server/blocks/discordblock.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+class DiscordReaderBlock(Block):
+    class Input(BlockSchema):
+        discord_bot_token: BlockSecret = SecretField(
+            key="discord_bot_token", description="Discord bot token"
+        )
+
+    class Output(BlockSchema):
+        message_content: str  # The content of the message received
+        channel_name: str  # The name of the channel the message was received from
+        username: str  # The username of the user who sent the message
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/blocks/discordblock.py]: aside: can we use Field() to have these populate with descriptions on the frontend?

File: rnd/autogpt_server/autogpt_server/util/settings.py [hunk 2]
```diff
@@ -105,6 +105,8 @@ class Secrets(UpdateTrackingModel["Secrets"], BaseSettings):
     medium_api_key: str = Field(default="", description="Medium API key")
     medium_author_id: str = Field(default="", description="Medium author ID")
 
+    discord_bot_token: str = Field(default="", description="Discord bot token")
+
     # Add more secret fields as needed
 
     model_config = SettingsConfigDict(
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/util/settings.py:112]: update env template

**Review by ntindle — CHANGES_REQUESTED** [review]

File: rnd/autogpt_server/autogpt_server/blocks/discordblock.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+            raise ValueError(f"An error occurred: {e}")
+
+
+class DiscordMessageSenderBlock(Block):
+    class Input(BlockSchema):
+        discord_bot_token: BlockSecret = SecretField(
+            key="discord_bot_token", description="Discord bot token"
+        )
+        channel_name: str  # The name of the channel to send the message to
+        message_content: str  # The content of the message to send
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/blocks/discordblock.py]: swap the order of these so they match the other one 
image

**Review by ntindle — CHANGES_REQUESTED** [review]

**Review by ntindle — APPROVED** [review]

**Pull request merged**